indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -499,14 +506,25 @@ private void syncFromStorage()
     try {
       if (active) {
         final List<Task> newTasks = taskStorage.getActiveTasks();
+        List<Task> addedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(newTasks), Sets.newHashSet(tasks)));
```

Comment:
AbstractTask defines equals/hashCode solely in terms of the id. It's meant to be a unique identifier.